gitconfig/gitconfig.sublime-syntax

```diff
@@ -0,0 +1,163 @@
+%YAML 1.2
+---
+# Syntax based on documentation here:
+# https://git-scm.com/docs/git-config#_syntax
+
+name: gitconfig
+file_extensions:
+  - gitconfig                 # /etc/gitconfig
+  - .gitconfig                # ~/.gitconfig
+first_line_match: ^\[core\]   # .git/config files always start with [core]
+scope: source.gitconfig
+
+variables:
+  section_name: '[\w.-]+(?=[\s\]])'
+  space_or_eol: (?=(?:\s|$))  # makes values on very last line match properly
+  variable_name: '[a-zA-Z][\w-]*'
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: section
+    - include: key-value-pair
+
+  boolean:
+    # 0 and 1 omitted due to mismatches (e.g. "log -1", "HEAD~1"). will be
+    # matched as unquoted strings instead
+    - match: \b(?:yes|true|on|off|no|false){{space_or_eol}}
+      scope: constant.language.gitconfig
+
+  color-attribute:
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink){{space_or_eol}}
+      scope: support.constant.color-attribute.gitconfig  # TODO: correct scope?
+
+  color-hex:
+    - match: (#)\h{6}
+      captures:
+        0: constant.other.color.rgb-value.gitconfig
+        1: punctuation.definition.constant.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  comments:
+    - match: (#|;).*
+      captures:
+        1: punctuation.definition.comment.gitconfig
+      scope: comment.line.gitconfig
+
+  # '\b', '\n', '\t', '\"', '\\'
+  escape:
+    - match: \\[\\"bnt]
+      scope: constant.character.escape.gitconfig
+
+  illegal-newline:
+    - match: \n
+      scope: invalid.illegal.gitconfig
+      pop: true
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: meta.key.gitconfig variable.language.gitconfig  # TODO: correct scope?
+        2: punctuation.separator.key-value.gitconfig
+      push:
+        # Hex colors like #efefef will be matched as comments unless we lower
+        # the precedence of comments. Here we disable the comments prototype and
+        # match comments after color-hex in the value rule.
+        - meta_include_prototype: false
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=\[)                            # start of new section
+          pop: true
+        - include: value
+
+  # [section-name "subsection"]
+  section:
+    - match: \[
+      scope: punctuation.definition.section.begin.gitconfig
```

Comment:
punctuation.section.brackets.begin ?

Reply:
For consistency with the scopes in the existing packages, I think `punctuation.definition.section.` makes more sense. For example, the `$` in PHP vars like `$foo` is scoped as `punctuation.definition.variable`, not `punctuation.definition.dollar`. But I'm happy to change it to `brackets` if you think it's wise. I'm 100% not attached to any of my scope names 🙂.